rubix-bookkeeper/src/main/java/com/qubole/rubix/bookkeeper/BookKeeper.java

```diff
@@ -423,18 +449,24 @@ public ClusterManager getClusterManagerInstance(ClusterType clusterType, Configu
   }
 
   @Override
-  public void setAllCached(String remotePath, long fileLength, long lastModified, long startBlock, long endBlock)
+  public void setAllCached(String remotePath, long fileLength, long lastModified, long startBlock, long endBlock, int generationNumber)
       throws TException
   {
     FileMetadata md;
     md = fileMetadataCache.getIfPresent(remotePath);
 
     //md will be null when 2 users try to update the file in parallel and both their entries are invalidated.
+    // different generation number means invalidation has occurred
     // TODO: find a way to optimize this so that the file doesn't have to be read again in next request (new data is stored instead of invalidation)
     if (md == null) {
       log.warn(String.format("Could not update the metadata for file %s", remotePath));
       return;
     }
+    if (md.getGenerationNumber() != generationNumber)
+    {
+      log.warn(String.format("Could not update the metadata for file %s having different generationNumber %d in cache and %d in request", remotePath, md.getGenerationNumber(), generationNumber));
```

Comment:
mark the log as debug level. We dont need to bring this to focus as this is a valid case